### PR TITLE
fix(fsapp): fix typing for app router

### PR DESCRIPTION
**Pull request by andyschwob**
In my original PR there were a lot of `any`s so I added some typing and also moved some types from the flagship-demo repo in here where it should be.

### CODE REVIEW

File: packages/fsapp/src/lib/app-router.ts [hunk 1]
```diff
@@ -302,7 +303,10 @@ export default class AppRouter {
         }
 
         return customRoutes.filter((page: PublishedPage) => page.path !== null)
-          .reduce((ret: any, page: any) => {
+          .reduce((ret: any, page: PublishedPage) => {
```
Comment (wSedlacek) [inline packages/fsapp/src/lib/app-router.ts:306]: Still noticing an `any` here that relates back to line 263 `async fetchPageRoutes(): Promise<any>`

From what I can tell this should be typed `Recored<string, Pick<PublishedPage , 'id' | 'title'>`

File: packages/fsapp/src/lib/app-router.ts [hunk 2]
```diff
@@ -273,12 +274,12 @@ export default class AppRouter {
           Authorization: `Bearer ${this.cmsToken}`
         }
         // tslint:disable-next-line: cyclomatic-complexity
-      }).then(async (pageData: any) => {
+      }).then(async (pageData: AxiosResponse) => {
```
Comment (wSedlacek) [inline packages/fsapp/src/lib/app-router.ts:277]: I would personally prefer that types that can be inferred, like callback types from promises, to be inferred. This tends to feel more natural for me and allows `.get()` to use a generic to control this type.

**Review by wSedlacek — CHANGES_REQUESTED** [review]
So much easier to read with the types being defined! There were a few that we still haven't defined types for yet but this is so much better.

File: packages/fsapp/src/lib/app-router.ts (hunk 1 above)
Comment (andyschwob) [reply packages/fsapp/src/lib/app-router.ts:306]: @wSedlacek changing this now, but i believe it should be:
`Record<string, Omit<PublishedPage, 'id' | 'path'>>,`

Comment (wSedlacek) [reply packages/fsapp/src/lib/app-router.ts:306]: `Omit` would exclude everything but the `id` and `path` where as on lines 310-312 it seems that we are defining `id` and `title`. That said I do see that `id`is renamed to `pageId` on those lines so the type I provided wasn't quite right on that aspect.

Sounds like you are on the right track though.

**Review by wSedlacek — APPROVED** [review]
LGTM

**Comment by wSedlacek** [comment]
@bweissbart Everything looks good here, feel free to merge if you are happy with it.

**Review by bweissbart — APPROVED** [review]

**Pull request merged**